provider/infoblox/infoblox.go

```diff
@@ -62,11 +63,15 @@ type StartupConfig struct {
 	NameRegEx     string
 	CreatePTR     bool
 	CacheDuration int
+	TenantId      string
+	CloudApiOwned string
```

Comment:
```suggestion
	CloudApiOwned bool
```

Reply:
I don't have an option if I carry the boolean type all the way to this struct, to know if a "false" value was actually set as an argument, or if its merely the default value of the type. I want to avoid including this attribute to every request with its default value. I'm currently thinking what my options are and suggestions are welcome too.

Reply:
What is the Infoblox default value for the option?

Reply:
By default the extensible attributes are not being set at all. They will only be set when you explicitly add the key/value to your request.

A simplified POST request would look like:

`curl -k -u user:pass -X POST "https://infoblox/wapi/v2.9.7/record:host?_return_fields%2B=name,extattrs" -d '{"name":"record","ipv4addrs": [{"ipv4addr":"1.1.1.1"}], "extattrs": {"Tenant ID": {"value": "MyTenant"}, "CMP Type": {"value": "Azure"}, "Cloud API Owned":{"value":"True" },"VM ID": { "value": "MYID"},"VM Name": { "value": "MyVM" }}}'`

There are more extensible attributes available, but they will not be set unless you add them yourself to the reuqest.

Reply:
Is there a situation in which one would configure some but not all three of these settings? Can sending the Cloud API Owned setting be conditional on, say, the tenant ID being configured?

In which situations would one send "True" and in which others would one send "False"?

Reply:
Is there a downside to sending "False" by default?

Reply:
Currently all three attributes are mandatory to create a record through the cloud api. So there is a reasonable case to automatically set the "Cloud API Owned" setting to "true" when the other two flags are set. I have played with that idea as well but I think there's some drawbacks as well. 

We don't know how infoblox will treat these attributes in the future and as a user it could be confusing at first that you can only set two out of three mandatory attributes.

Sending "False" by default would mean that all records managed by the infoblox provider will get an extra attribute "Cloud API Owned=False". Normally that shouldn't break anything, but I also cannot be for sure that somebody has some logic build around this field. Updating dns records with this field just to make it work doesn't seem like a good idea to me.

So from what I see I think we can either keep the value a string at the front, and let users explicitly decide whether they want this field set or not, or we should remove this field and infer it from the other two flags that are mandatory for the cloud api. What do you think?

Reply:
It seems to me that external-dns owns the records that it is creating. The question then becomes whether external-dns counts as "Cloud API" for purposes of that attribute. External-dns may well run on Azure, but it is not Azure itself.

It seems likely to me that external-dns should either always set it to "True" or always set it to "False" when a TenantID is specified, depending on what the semantics of the attribute actually are. But since neither of us can seem to figure it out, I don't see how someone configuring external-dns could either. I'm not inclined to accept this PR until we have more definitive information from someone who knows what the semantics of these fields actually are.